probes/codeReviewTwoReviewers/impl.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/utils"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const probe = "codeReviewTwoReviewers"
+const minimumReviewers = 2
+var reviewerLoginErr = fmt.Errorf("could not find the login of a reviewer")
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	rawReviewData := &raw.CodeReviewResults
+	return codeReviewRun(rawReviewData, fs, probe, finding.OutcomePositive, finding.OutcomeNegative)
+}
+
+/*
+** Looks through the data and validates author and reviewers of a changeset
+** Scorecard currently only supports GitHub revisions and generates a positive
+** score in the case of other platforms. This probe is created to ensure that
+** there are a number of unique reviewers for each changeset.
+ */
+
+func codeReviewRun(reviewData *checker.CodeReviewData, fs embed.FS, probeID string,
+	positiveOutcome, negativeOutcome finding.Outcome,
+) ([]finding.Finding, string, error) {
+	changesets := reviewData.DefaultBranchChangesets
+	var findings []finding.Finding
+	leastFoundReviewers := 0
+	numChangesets := len(changesets)
+	numBotAuthors := 0
+	if numChangesets == 0 {
+		return nil, probeID, utils.NoChangesetsErr
+	}
+	// Loops through all changesets, if an author login cannot be retrieved: returns OutcomeNotAvailabe.
+	// leastFoundReviewers will be the lowest number of unique reviewers found among the changesets.
+	for i := range changesets {
+		data := &changesets[i]
+		if data.Author.Login == "" {
+			f, err := finding.NewNotAvailable(fs, probeID, "Could not retrieve the author of a changeset.", nil)
+			if err != nil {
+				return nil, probeID, fmt.Errorf("create finding: %w", err)
+			}
+			findings = append(findings, *f)
+			return findings, probeID, nil
+		} else if data.Author.IsBot {
+			numBotAuthors += 1
+		}
+		numReviewers, err := uniqueReviewers(data.Author.Login, data.Reviews)
+		if err != nil {
+			f, err := finding.NewNotAvailable(fs, probeID, "Could not retrieve the reviewer of a changeset.", nil)
+			if err != nil {
+				return nil, probeID, fmt.Errorf("create finding: %w", err)
+			}
+			findings = append(findings, *f)
+			return findings, probeID, nil
+		} else if i == 0 || numReviewers < leastFoundReviewers {
+			leastFoundReviewers = numReviewers
+		}
+	}
+	if numBotAuthors == numChangesets {
+		// returns a NotAvailable outcome if all changesets were authored by bots
+		f, err := finding.NewNotAvailable(fs, probeID, "All changesets authored by bot(s).", nil)
+		if err != nil {
+			return nil, probeID, fmt.Errorf("create finding: %w", err)
+		}
+		findings = append(findings, *f)
+		return findings, probeID, nil
+	} else if leastFoundReviewers >= minimumReviewers {
+		// returns PositiveOutcome if the lowest number of unique reviewers is at least as high as minimumReviewers (2).
+		f, err := finding.NewWith(fs, probeID, fmt.Sprintf("%v unique reviewers found for at least one " +
```

Comment:
Should this message be phrased as ">%d unique reviewers found for all changesets"?